models/badges.js

```diff
@@ -27,6 +28,35 @@ const fetchBadges = async ({
   }
 }
 
+/**
+ * Fetches the data about user badges
+ * @param query { Object }: Filter for badges data
+ * @return {Promise<userBadgeModel|Array>}
+ */
+
+const fetchUserBadges = async (username) => {
+  try {
+    const userBadges = []
+    let userExists = false
+    const result = await fetchUser({ username })
+    if (result.userExists) {
+      userExists = true
+      const userID = result.user.id
+      const snapshot = await badgeModel.get()
+      snapshot.forEach((item) => {
+        if (item.data().users.includes(userID)) {
```

Comment:
Better to use optional chaining here? What if some item doesn't have data property? What if some data property doesn't have users? Could happen, right? API will break in that case

Reply:
Okay will add that